books/free-programming-books-zh.md

```diff
@@ -421,7 +421,7 @@
 
 ### Groovy
 
-* [实战 Groovy 系列](http://www.ibm.com/developerworks/cn/java/j-pg/)
+* [实战 Groovy 系列](https://wizardforcel.gitbooks.io/ibm-j-pg/content) - WizardForcel
```

Comment:
"WizardForce" is not the author. Why do we think this is an authorized distribution?

Reply:
![image](https://user-images.githubusercontent.com/3125580/139562802-63d445c8-dfef-49ff-a1a4-ada4c4995af8.png)

Maybe come back in a future

Last backup from previous resource: https://web.archive.org/web/20181023171953/http://www.ibm.com/developerworks/cn/java/j-pg/

Reply:
So let's use the backup one or just keep the original one?

Reply:
Also consider maybe deleting it?

Reply:
I found this might work: [Groovy 教程](https://www.w3cschool.cn/groovy/) since some of our resources are W3Cschool, too.

If this is ok, I will submit the change.

Reply:
ok

Reply:
The reason to post the backup link and screenshot is to have some context about,

Since is your mother language you can decide better which content is more valuable